template/src/ontology/run.sh.jinja2

```diff
@@ -45,21 +56,30 @@ if [ -n "$ODK_BINDS" ]; then
     VOLUME_BIND="$VOLUME_BIND,$ODK_BINDS"
 fi
 
+# Gather environment variables into a single file
+cat <<EOF > run.sh.env
+ROBOT_JAVA_ARGS=$ODK_JAVA_OPTS
+JAVA_OPTS=$ODK_JAVA_OPTS
+GH_TOKEN=$GH_TOKEN
+EOF
+
 if [ -n "$USE_SINGULARITY" ]; then
     {% if project.use_external_date is sameas True %}export TODAY=$(date +%Y-%m-%d){% endif %}
     singularity exec --cleanenv $ODK_SINGULARITY_OPTIONS \
-        --env "ROBOT_JAVA_ARGS=$ODK_JAVA_OPTS,JAVA_OPTS=$ODK_JAVA_OPTS" \
+        --env-file run.sh.env \
         --bind $VOLUME_BIND \
         -W $WORK_DIR \
         docker://obolibrary/$ODK_IMAGE:$ODK_TAG $TIMECMD "$@"
 else
     BIND_OPTIONS="-v $(echo $VOLUME_BIND | sed 's/,/ -v /')"
     docker run -u $(id -u):$(id -g) $ODK_DOCKER_OPTIONS $BIND_OPTIONS -w $WORK_DIR \
-        -e ROBOT_JAVA_ARGS="$ODK_JAVA_OPTS" -e JAVA_OPTS="$ODK_JAVA_OPTS" \
+        --env-file run.sh.env \
```

Comment:
You do this because it is a security to risk to pass the token with -e right?

Reply:
At least it’s a _theoretical_ security risk.

To be honest, I think it’s borderline security theatre. If an attacker has enough access to your machine to be able to read a process’ command line, you already have a serious security issue and your GitHub token (as well as any other credential you may have stored on your machine) is already at risk – making sure it does not appear in a command line will not save you.

I did it that way because I feel some people might be reluctant to use the feature if they see their GitHub token being used on a command line. But you wouldn’t have to push me too hard to convince me that it is not worth it.

Reply:
No I think I agree with it and it sets us up for future similar situations.